_posts/2020-05-01-native-command-mode-tools.adoc

```diff
@@ -0,0 +1,137 @@
+---
+layout: post
+title: 'Ideas on Quarkus command-mode, UI-based apps and releasing via Github Actions'
+date: 2020-05-10
+tags: development-tips
+synopsis: Ideas on how to use Quarkus command-mode to develop native applications with web frontends; built with Github Actions.
+author: dplatz
+---
+:imagesdir: /assets/images/posts/quarkus-native-app
+
+Quarkus is advertised as a cloud-native runtime for Microservices but the link:https://quarkus.io/blog/quarkus-1-4-final-released/[introduction of command-mode] might have opened up new and intresting spaces where it could find adoption.
+
+In this blog post I will present a few ideas about the possible usage of command-mode in Quarkus that you might find useful to consider when developing your next tool.
+If you have not heard about command-mode, I suggest to start reading link:https://quarkus.io/blog/introducing-command-mode/[this introduction] first and then come back.
+
+Ready? Ok, let's continue.
+
+== Cross-platform builds via Github Actions
+
+Command-line tools like `kubectl` are often written in languages like C, Go or Rust and compiled to native executables for various target platforms.
+This is great because you don't need to rely on Java installed; or beeing installed in a wrong version.
+
+With GraalVM native-images, this argument against Java does not hold anymore. Quarkus supports the build of native executables via GraalVM. Using it in combination with command-mode sounds a very productive choice to write command-line tools.
```

Comment:
might want to mention that Quarkus supports mainly building natively for Linux containers, but also has experimental support for OS X and Windows or something like that

Reply:
Rephrased it to:
With GraalVM native-images, this argument against Java does not hold anymore. Quarkus supports the build of native executables via GraalVM; mainly for building natively for Linux containers but there is experimental support for OS X and Windows as well.
Using the native build in combination with command-mode sounds like a very interesting and productive choice to write command-line tools.